src/library/format/qubist/parser.jl

```diff
@@ -0,0 +1,83 @@
+function _parse_entry!(data::Dict{Symbol,Any}, line::AbstractString, ::Qubist)
+    m = match(r"^\s*([0-9]+)\s+([0-9]+)\s+([+-]?([0-9]*[.])?[0-9]+)\s*$", line)
+
+    if isnothing(m)
+        return false
+    end
+
+    i = tryparse(Int, m[1])
+    j = tryparse(Int, m[2])
+    c = tryparse(Float64, m[3])
+
+    if isnothing(i) || isnothing(j) || isnothing(c)
+        syntax_error("Invalid entry: '$line'")
+    elseif i == j
+        L    = data[:linear_terms]
+        L[i] = get(L, i, 0.0) + c
+    else
+        Q         = data[:quadratic_terms]
+        Q[(i, j)] = get(Q, (i, j), 0.0) + c
+    end
+
+    return true
+end
+
+function _parse_header!(data::Dict{Symbol,Any}, line::AbstractString, ::Qubist)
+    m = match(r"^([0-9]+)\s+([0-9]+)$", line)
+
+    if isnothing(m)
+        return false
+    end
+
+    dimension  = parse(Int, m[1])
+    total_size = parse(Int, m[2])
+
+    if isnothing(dimension) || isnothing(total_size)
+        syntax_error("Invalid header: '$line'")
+    end
+
+    data[:dimension] = dimension
+
+    # TODO: `sizehint!` linear and quadratic collections
+    # IDEA: 
+    #   1. linear_size + quadratic_size = total_size
+    #   2. dimension * linear_size    ≈ linear_size + 2 quadratic_size
+    #   
+    #   3. => quadratic_size = (dimension - 1) * linear_size ÷ 2 [2]
+    #   4. => linear_size    ≈ 2 * total_size ÷ (dimension + 1)  [1, 3]
```

Comment:
What would this hint be useful for? The math checks out

Reply:
Dict pre-allocation

Reply:
Maybe add a comment